spec/requests/api/v1/garden_request_spec.rb

```diff
@@ -0,0 +1,141 @@
+        expect(garden[:attributes][:name]).to eq(new_garden.name)
+      end
+
+      it 'can create a new garden' do
+        garden_params = {longitude: 100.5, latitude: 97.5, name: "Garden 1", description: "it's a garden, what else do you want to know?"}
+
+        headers = {"CONTENT_TYPE" => "application/json"}
+        post "/api/v1/gardens", headers: headers, params: JSON.generate(garden_params)
+
+        garden = Garden.last
+
+        expect(response).to be_successful
+        expect(garden.longitude).to eq(garden_params[:longitude])
+        expect(garden.latitude).to eq(garden_params[:latitude])
+        expect(garden.name).to eq(garden_params[:name])
+      end
+
+      it 'can update an existing garden' do
+        garden = create(:garden)
+        garden_params = {longitude: 100.5, latitude: 97.5, name: "Other Name", description: "it's a garden yo"}
+
+        headers = { 'CONTENT_TYPE' => 'application/json' }
+        patch "/api/v1/gardens/#{garden.id}", headers: headers, params: JSON.generate(garden_params)
+        expect(response).to be_successful
+        updated_garden = Garden.find_by(id: garden.id)
+
+        expect(garden.description).to eq(nil)
+        expect(updated_garden.longitude).to_not eq(garden.longitude)
+        expect(updated_garden.latitude).to_not eq(garden.latitude)
+        expect(updated_garden.name).to_not eq(garden.name)
+
+        expect(updated_garden.longitude).to eq(garden_params[:longitude])
+        expect(updated_garden.latitude).to eq(garden_params[:latitude])
+        expect(updated_garden.name).to eq(garden_params[:name])
+        expect(updated_garden.description).to eq(garden_params[:description])
+      end
+
+      it 'can destroy a garden' do
+        garden = create(:garden)
+
+        expect(Garden.count).to eq(1)
+        delete "/api/v1/gardens/#{garden.id}"
+        expect(response).to be_successful
+        expect(Garden.count).to eq(0)
+        expect{Garden.find(garden.id)}.to raise_error(ActiveRecord::RecordNotFound)
+      end
+  end
+
+  describe 'sad paths' do
+   it 'index - returns a 204 if query entered wrong' do
+     get '/api/v1/gardens'
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+
+   it 'show - returns a 204 if query entered wrong' do
+     get "/api/v1/gardens/99999"
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+
+   it 'create - returns a 204 if query entered wrong' do
+     post "/api/v1/gardens"
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+
+   it 'update - returns a 204 if query entered wrong' do
+     patch "/api/v1/gardens/999999"
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+
+   it 'delete - returns a 204 if query entered wrong' do
+     delete "/api/v1/gardens/999999"
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+  end
+end
```

Comment:
I had a brief conversation with @AngelaGuardia about returning a 204 No Content. While I think it's a great idea to do sad-path testing and account for incorrect data being sent, I believe later we should revise the controller to send back the proper HTTP code based on the [JSON Specification Guide](https://jsonapi.org/format/). Good for now I think.

Reply:
Do you think we should make an extension or another issue that explicitly deals with the 204 error?